src/sentry/auth/superuser.py

```diff
@@ -85,7 +87,24 @@ def get_superuser_scopes(auth_state: RpcAuthState, user: Any):
     return superuser_scopes
 
 
-def is_active_superuser(request):
+def superuser_has_permission(request: HttpRequest | Request):
+    if not is_active_superuser(request):
+        return False
+
+    if is_self_hosted():
+        return True
+
+    if features.has("auth:enterprise-superuser-read-write", actor=request.user):
+        if "superuser.write" in request.access.permissions:
+            return True
+
+        # superuser read-only can only hit GET and OPTIONS (pre-flight) requests
+        return request.method == "GET" or request.method == "OPTIONS"
+
+    return True
```

Comment:
Should the default for this be `False`? It's safer for the user to not be a superuser, and only evaluate to true when the right conditions are met. We want to keep the security risk minimized right?

Reply:
we already check and return early in the function if the user is not an active superuser, and that function also returns false if the user can't become a superuser anyway.

if we get to the line where we check if the user has the flag, then they are active superuser. if they don't have the flag, then we aren't enforcing the read-write separation so we always give them permission. i can see where this is a bit confusing though. a comment might help?

Reply:
You could also just restructure the if statements

rather than returning early, you could just wrap everything in an 
```
if is_active_superuser():
    ...
return False
```

🤷

Reply:
ahhh i get it, I think this might help then:

```suggestion
   #if the user is not FF gated, they are active superuser with permissions
    if not features.has("auth:enterprise-superuser-read-write", actor=request.user):
        return True
        
    #otherwise, check based on the read/write level 
    if "superuser.write" in request.access.permissions:
            return True

    # superuser read-only can only hit GET and OPTIONS (pre-flight) requests
    return request.method == "GET" or request.method == "OPTIONS"
```

Reply:
you could even combine the 3 true checks with ors

Reply:
ok i have some refactoring to do 😅